lib/api/v3/user_preferences/user_preferences_representer.rb

```diff
@@ -0,0 +1,80 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'roar/decorator'
+require 'roar/json/hal'
+
+module API
+  module V3
+    module UserPreferences
+      class UserPreferencesRepresenter < ::API::Decorators::Single
+        link :self do
+          {
+            href: api_v3_paths.my_preferences
+          }
+        end
+
+        link :user do
+          {
+            href: api_v3_paths.user(current_user.id),
+            title: current_user.name
+          }
+        end
+
+        link :updateImmediately do
+          {
+            href: api_v3_paths.my_preferences,
+            method: :patch
+          }
+        end
+
+        property :hide_mail
+        property :time_zone,
+                 getter: -> (*) { canonical_time_zone },
+                 render_nil: true
+
+        property :theme
+        property :comments_in_reverse_order?,
+                 as: :commentSortDescending,
+                 setter: -> (value, *) { self.comments_in_reverse_order = value }
+
+        property :warnOnLeavingUnsaved,
+                 getter: -> (*) { warn_on_leaving_unsaved? },
+                 setter: -> (value, *) { self.warn_on_leaving_unsaved = value }
```

Comment:
Since you added the builder aliases the two properties above should be able to become one-liners (did not test it, though):

````ruby
property :comments_in_reverse_order, as: :commentSortDescending # okay, the "as:" might make it two lines...
property :warn_on_leaving_unsaved # camelization should happen automatically
````